Update documentation to explain behavior in a Catch-all route (#89742)
vercel/next.js: docs/01-app/03-api-reference/04-functions/use-selected-layout-segment.mdx

>
> - Since `useSelectedLayoutSegment` is a [Client Component](/docs/app/getting-started/server-and-client-components) hook, and Layouts are [Server Components](/docs/app/getting-started/server-and-client-components) by default, `useSelectedLayoutSegment` is usually called via a Client Component that is imported into a Layout.
> - `useSelectedLayoutSegment` only returns the segment one level down. To return all active segments, see [`useSelectedLayoutSegments`](/docs/app/api-reference/functions/use-selected-layout-segments)

> > - For [Catch-all](/docs/app/api-reference/file-conventions/dynamic-routes#catch-all-segments) routes like `[...slug]`, visiting `/blog/a/b/c` returns `'a/b/c'` as a single string, not separate segments.

Reviewer: I've been thinking about whether, while helpful, this still leaves some questions open

```suggestion
> - For [catch-all](/docs/app/api-reference/file-conventions/dynamic-routes#catch-all-segments) routes, the matched segments are returned as a single joined string. For example, given `app/blog/[...slug]/page.js`, calling from `app/blog/layout.js` when visiting `/blog/a/b/c` returns `'a/b/c'`.
```

That kind of suggestion is likely a better statement here

> > - `useSelectedLayoutSegment` reflects the file-system route **not** the pathname, therefore in a [Catch-all](/docs/app/api-reference/file-conventions/dynamic-routes#catch-all-segments) route it will return all segments as a single string instead of only the segment one level down.

Reviewer: ```suggestion
> For [Catch-all](/docs/app/api-reference/file-conventions/dynamic-routes#catch-all-segments) routes like `[...slug]`, visiting `/blog/a/b/c` returns `'a/b/c'` as a single string, not separate segments.
```

Or is not, maybe something saying `the catch-all parameter value is returned as a single string joined with /` 

I think the suggestion is clear and concrete though
